### PR TITLE
[CM-1016] feat: Commerce Widget Name + Soft Release

### PR DESCRIPTION
**Pull request by jwhardwick**
### Hi👋, please prefix this PR's title with:

- [ ] `breaking-change:` if you have introduced modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities.
- [x] `feat:`, `fix:`, `refactor:`, `docs:`, or `chore:`.

* Rename Checkout Widget -> Immutable Commerce Widget
* Soft release Commerce Widget

### CODE REVIEW

File: packages/checkout/widgets-lib/src/factory.ts [hunk 1]
```diff
@@ -95,14 +94,11 @@ export class WidgetsFactory implements IWidgetsFactory {
           provider,
         }) as Widget<WidgetType.SALE> as Widget<T>;
       }
-      case WidgetType.CHECKOUT: {
-        if (this.sdk.config.environment === Environment.PRODUCTION) {
-          throw new Error('Checkout widget is not supported in production');
-        }
-        return new CheckoutWidgetRoot(this.sdk, {
+      case WidgetType.IMMUTABLE_COMMERCE: {
+        return new CommerceWidgetRoot(this.sdk, {
```
Comment (jwhardwick) [inline packages/checkout/widgets-lib/src/factory.ts:98]: NOTE -> this will soft release this widget to prod.

Comment (jwhardwick) [reply packages/checkout/widgets-lib/src/factory.ts:98]: maybe use a `https://tsdoc.org/pages/tags/beta/` decorator here.

File: packages/checkout/widgets-lib/tsconfig.json [hunk 2]
```diff
@@ -11,6 +11,6 @@
     "jsxImportSource": "@emotion/react",
     "experimentalDecorators": true,
   },
-  "include": ["src"],
+  "include": ["src", "src/widgets/unified-commerce/.tsx"],
```
Comment (jwhardwick) [inline packages/checkout/widgets-lib/tsconfig.json:14]: remove.

File: packages/checkout/sdk/src/widgets/definitions/types.ts [hunk 3]
```diff
@@ -357,11 +355,11 @@ export type SemanticVersion = {
  * @property {boolean | undefined} isBridgeEnabled
  */
 export type CheckoutWidgetsConfig = {
-  /** The theme of the Checkout Widget (default: "DARK") */
+  /** The theme of the Commerce Widget (default: "DARK") */
```
Comment (jwhardwick) [inline packages/checkout/sdk/src/widgets/definitions/types.ts:358]: todo - don't need to change it here

File: packages/checkout/sdk/src/widgets/load.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { useLocalBundle } from '../env';
 
-// Loads the checkout widgets bundle from the CDN and appends the script to the document head
+// Loads the Commerce Widgets bundle from the CDN and appends the script to the document head
```
Comment (jwhardwick) [inline packages/checkout/sdk/src/widgets/load.ts:3]: todo - don't need to change it ehre

**Comment by jwhardwick** [comment]
We expect `PR / Build, Lint & Test Example Widgets with latest SDK` to fail here.

**Review by jhesgodi — APPROVED** [review]

**Review by zaidarain1 — DISMISSED** [review]

File: packages/checkout/widgets-lib/src/factory.ts (hunk 1 above)
Comment (jwhardwick) [reply packages/checkout/widgets-lib/src/factory.ts:98]: It's fine as is.

**Review by zaidarain1 — COMMENTED** [review]
Do we still need the 
```
build-lint-test-example-widgets-with-sdk:
    name: Build, Lint & Test Example Widgets with latest SDK
```
workflow step in the pr.yaml?

**Comment by jwhardwick** [comment]
> Do we still need the
> 
> ```
> build-lint-test-example-widgets-with-sdk:
>     name: Build, Lint & Test Example Widgets with latest SDK
> ```
> 
> workflow step in the pr.yaml?

@zaidarain1 

It's getting removed in this PR https://github.com/immutable/ts-immutable-sdk/pull/2346. Both of these PRs have conflicts which each other, I'll have to resolve once one is merged.

**Comment by jwhardwick** [comment]
https://github.com/immutable/ts-immutable-sdk/actions/runs/11491972155/job/31985265313?pr=2349 failing as expected, need to commit to checkout-api-configs.

**Review by jhesgodi — DISMISSED** [review]

**Review by zaidarain1 — APPROVED** [review]

**Review by jhesgodi — APPROVED** [review]

**Pull request merged**